src/main.rs

```diff
@@ -70,36 +49,42 @@ fn main() {
 }
 
 fn run() -> Result<(), MainError> {
-    let args: Vec<String> = std::env::args().collect();
-    match &*args[1] {
+    let cli = Cli::parse();
+    match &*cli.name[0] {
         "run" | "run-script" => {
-            let manifest = "package.json"
+            let mut cwd = env::current_dir().expect("Couldn't find the current working directory");
+            if cli.workspace_root {
+                cwd = workspace::find_workspace_root(&cwd)?;
+            }
+            let manifest_path = cwd.join("package.json");
+            let manifest = manifest_path
                 .pipe(File::open)
                 .map_err(MainError::from_dyn)?
                 .pipe(serde_json::de::from_reader::<_, NodeManifest>)
                 .map_err(MainError::from_dyn)?;
-            let name = args[2].clone();
+            let name = cli.name[1].clone();
             if let Some(command) = manifest.scripts.get(&name) {
                 eprintln!("> {:?}", command);
-                run_script(name, command.clone())
+                run_script(name, command.clone(), &cwd)
             } else {
                 PnError::MissingScript { name }
                     .pipe(MainError::Pn)
                     .pipe(Err)
             }
         }
-        "install" | "i" | "update" | "up" => pass_to_pnpm(&args[1..]),
-        _ => pass_to_sub(args[1..].join(" ")),
+        "install" | "i" | "update" | "up" => pass_to_pnpm(&cli.name[0..]),
+        _ => pass_to_sub(cli.name[0..].join(" ")),
     }
 }
 
-fn run_script(name: String, command: String) -> Result<(), MainError> {
+fn run_script(name: String, command: String, cwd: &dyn AsRef<Path>) -> Result<(), MainError> {
```

Comment:
I will often just do this:

```suggestion
fn run_script(name: String, command: String, cwd: &Path) -> Result<(), MainError> {
```

which would sometimes requires you to explicitly call `.as_ref()` for non-path-deref types (e.g. a `String` or a `str`).

If you dislike explicitly calling `.as_ref()`, you can do this:

```suggestion
fn run_script(name: String, command: String, cwd: impl AsRef<Path>) -> Result<(), MainError> {
```

which would allow you to pass `cwd` to either move `cwd` or merely borrow it, depends on whether you passed it by value or by reference at the call site.

And if you want to be able to specify the type of `cwd` explicitly, do this:

```suggestion
fn run_script<Cwd: AsRef<Path>>(name: String, command: String, cwd: Cwd) -> Result<(), MainError> {
```